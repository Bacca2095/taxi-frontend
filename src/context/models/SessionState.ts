export interface SessionState {
  document?: string;
  sessionId?: string;
}
