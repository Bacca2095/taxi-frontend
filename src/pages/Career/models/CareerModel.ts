/* eslint-disable prettier/prettier */
export interface CareerModel {
  id?: number;
  nombre: string;
  documento: string;
  telefono: number;
  fechaRecogida: string | Date;
  horaRecogida: string;
  direccion: string;
  costo?: number;
}
