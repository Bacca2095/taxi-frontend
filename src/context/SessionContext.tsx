/* eslint-disable @typescript-eslint/explicit-module-boundary-types */
import React, { createContext, useEffect, useRef, useState } from 'react';
import { SessionState } from 'context/models/SessionState';
import * as sessionService from '../services/sessionService';

export const useStateContainer = (
  initialState: SessionState = { document: '', sessionId: '' },
) => {
  const [document, setDocument] = useState(initialState.document);
  const [sessionId, setSessionId] = useState(initialState.sessionId);
  const documentRef = useRef(false);

  const clearSession = () => {
    sessionService.deleteSession();
  };

  useEffect(() => {
    if (!documentRef.current) {
      documentRef.current = true;
      return;
    }
    if (!document) {
      return;
    }
    setSessionId(sessionService.createSession(document));
  }, [document]);

  return {
    data: { document, sessionId },
    mutations: { setDocument, clearSession },
  };
};

export const SessionContext = createContext<
  ReturnType<typeof useStateContainer>
>({} as never);

export const SessionProvider: React.FC = ({ children }) => {
  const contextValue = useStateContainer(sessionService.recoverSession());
  return (
    <SessionContext.Provider value={contextValue}>
      {children}
    </SessionContext.Provider>
  );
};
