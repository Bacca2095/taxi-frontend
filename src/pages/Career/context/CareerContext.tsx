/* eslint-disable @typescript-eslint/explicit-module-boundary-types */
import React, {
  createContext,
  useContext,
  useEffect,
  useRef,
  useState,
} from 'react';
import { SessionContext } from 'context/SessionContext';
import * as careerService from '../services/careerService';
import { CareerModel } from '../models/CareerModel';

export interface CareerState {
  careers?: CareerModel[];
}

export const useStateContainer = (
  initialState: CareerState = {},
  sessionContext: typeof SessionContext,
) => {
  const {
    data: { sessionId },
  } = useContext(sessionContext);
  const [allCareers, setAllCareers] = useState(initialState.careers || []);
  const [currentCareerId, setCurrentCareerId] = useState(0);
  const [errorOnDelete, setErrorOnDelete] = useState(false);
  const [errorOnCreate, setErrorOnCreate] = useState(false);
  const [updateList, setUpdateList] = useState(false);
  const careerRef = useRef(false);

  const deleteCareer = () => {
    careerService.deleteCareer(currentCareerId).then((res) => {
      if (res) {
        setErrorOnDelete(false);
      } else {
        setErrorOnDelete(true);
      }
      setUpdateList(!updateList);
    });
  };

  const createCareer = (career: CareerModel) => {
    careerService.createCareer(career).then((res) => {
      if (res) {
        setErrorOnCreate(false);
      } else {
        setErrorOnCreate(true);
      }

      setUpdateList(!updateList);
    });
  };

  useEffect(() => {
    if (!careerRef.current) {
      careerRef.current = true;
      return;
    }
    if (!document) {
      return;
    }
    careerService
      .listCareer(sessionId?.split(':::')[1])
      .then((career) => setAllCareers(career));
  }, [sessionId, updateList]);

  return {
    data: { allCareers, currentCareerId, errorOnDelete, errorOnCreate },
    mutations: {
      setCurrentCareerId,
      deleteCareer,
      createCareer,
    },
  };
};

export const CareerContext = createContext<
  ReturnType<typeof useStateContainer>
>({} as never);

export interface CareerProviderProps {
  initialState?: CareerState;
}
export const CareerProvider: React.FC<CareerProviderProps> = ({
  children,
  initialState = {},
}) => {
  const contextValue = useStateContainer(initialState, SessionContext);
  return (
    <CareerContext.Provider value={contextValue}>
      {children}
    </CareerContext.Provider>
  );
};
