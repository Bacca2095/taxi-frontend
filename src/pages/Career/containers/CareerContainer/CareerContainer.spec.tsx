import React from 'react';
import nock from 'nock';
import { render } from '@testing-library/react';
import { CareerProvider } from 'pages/Career/context/CareerContext';
import { SessionContext } from 'context/SessionContext';
import * as careerFixture from 'tests/fixtures/career.fixture';
import { CareerModel } from 'pages/Career/models/CareerModel';
import { CareerContainer } from '.';

describe('Career Container tests', () => {
  let setDocument: jest.Mock;
  let clearSession: jest.Mock;
  let careers: CareerModel[];
  const document = '12345';
  const sessionId = '12345:::12345';

  beforeAll(() => {
    setDocument = jest.fn();
    clearSession = jest.fn();
    careers = careerFixture.getList();
    nock('http://localhost:3001/api')
      .persist()
      .get('/carreras/12345')
      .reply(200, careers);
  });

  it('should match snapshot', () => {
    const { container } = render(
      <SessionContext.Provider
        value={{
          data: { document, sessionId },
          mutations: { setDocument, clearSession },
        }}
      >
        <CareerProvider>
          <CareerContainer />
        </CareerProvider>
      </SessionContext.Provider>,
    );
    expect(container).toMatchSnapshot();
  });
});
