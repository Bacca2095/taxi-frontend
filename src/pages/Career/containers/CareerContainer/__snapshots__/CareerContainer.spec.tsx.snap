// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Career Container tests should match snapshot 1`] = `
<div>
  <div
    class="MuiGrid-root MuiGrid-container MuiGrid-spacing-xs-3 MuiGrid-align-items-xs-center MuiGrid-justify-content-xs-center"
  >
    <div
      class="MuiGrid-root MuiGrid-item MuiGrid-grid-xs-12"
    >
      <div
        class="MuiGrid-root MuiGrid-container MuiGrid-align-items-xs-center MuiGrid-justify-content-xs-center"
      >
        <div
          class="MuiGrid-root MuiGrid-item MuiGrid-grid-xs-12 MuiGrid-grid-md-10"
        >
          <h4
            class="MuiTypography-root MuiTypography-h4"
          >
            Carreras
          </h4>
        </div>
        <div
          class="MuiGrid-root MuiGrid-item MuiGrid-grid-xs-12 MuiGrid-grid-md-2"
        >
          <div
            class="MuiGrid-root MuiGrid-container MuiGrid-justify-content-xs-center"
          >
            <div
              class="MuiGrid-root"
            >
              <button
                class="MuiButtonBase-root MuiIconButton-root MuiIconButton-colorPrimary"
                data-testid="create"
                tabindex="0"
                title="Crear Carrera"
                type="button"
              >
                <span
                  class="MuiIconButton-label"
                >
                  <svg
                    aria-hidden="true"
                    class="MuiSvgIcon-root MuiSvgIcon-fontSizeLarge"
                    focusable="false"
                    viewBox="0 0 24 24"
                  >
                    <path
                      d="M12 2C6.48 2 2 6.48 2 12s4.48 10 10 10 10-4.48 10-10S17.52 2 12 2zm4 11h-3v3c0 .55-.45 1-1 1s-1-.45-1-1v-3H8c-.55 0-1-.45-1-1s.45-1 1-1h3V8c0-.55.45-1 1-1s1 .45 1 1v3h3c.55 0 1 .45 1 1s-.45 1-1 1z"
                    />
                  </svg>
                </span>
                <span
                  class="MuiTouchRipple-root"
                />
              </button>
            </div>
            
          </div>
        </div>
      </div>
    </div>
    <div
      class="MuiGrid-root MuiGrid-item MuiGrid-grid-xs-12"
    >
      <div
        class="MuiGrid-root MuiGrid-container MuiGrid-spacing-xs-3"
      >
        <div
          class="MuiGrid-root makeStyles-dataGrid-1 MuiGrid-item MuiGrid-grid-xs-12"
        >
          <div
            aria-colcount="7"
            aria-multiselectable="false"
            aria-rowcount="0"
            class="MuiDataGrid-root MuiDataGrid-root"
            role="grid"
          >
            <div>
              <div />
            </div>
            <div
              class="MuiDataGrid-main"
            >
              <div
                class="MuiDataGrid-overlay"
                style="top: 56px;"
              >
                No rows
              </div>
              <div
                class="MuiDataGrid-columnsContainer"
                style="min-height: 56px; max-height: 56px; line-height: 56px;"
              >
                <div
                  aria-rowindex="1"
                  class="MuiDataGrid-columnHeaderWrapper scroll"
                  role="row"
                  style="transform: translate3d(-0px, 0, 0); min-width: 1010px;"
                >
                  <div
                    aria-colindex="1"
                    class="MuiDataGrid-columnHeader"
                    data-field="nombre"
                    role="columnheader"
                    style="width: 120px; min-width: 120px; max-width: 120px;"
                    tabindex="0"
                  >
                    <div
                      class="MuiDataGrid-columnHeaderDraggableContainer"
                      draggable="false"
                    >
                      <div
                        class="MuiDataGrid-columnHeaderTitleContainer"
                      >
                        <div
                          class="MuiDataGrid-columnHeaderTitle"
                          title=""
                        >
                          Nombre
                        </div>
                      </div>
                      <div
                        class="MuiDataGrid-menuIcon"
                      >
                        <button
                          aria-controls="mui-4671"
                          aria-haspopup="true"
                          aria-label="Menu"
                          class="MuiButtonBase-root MuiIconButton-root MuiDataGrid-menuIconButton MuiIconButton-sizeSmall"
                          id="mui-93070"
                          tabindex="-1"
                          title="Menu"
                          type="button"
                        >
                          <span
                            class="MuiIconButton-label"
                          >
                            <svg
                              aria-hidden="true"
                              class="MuiSvgIcon-root MuiSvgIcon-fontSizeSmall"
                              focusable="false"
                              viewBox="0 0 24 24"
                            >
                              <path
                                d="M12 8c1.1 0 2-.9 2-2s-.9-2-2-2-2 .9-2 2 .9 2 2 2zm0 2c-1.1 0-2 .9-2 2s.9 2 2 2 2-.9 2-2-.9-2-2-2zm0 6c-1.1 0-2 .9-2 2s.9 2 2 2 2-.9 2-2-.9-2-2-2z"
                              />
                            </svg>
                          </span>
                          <span
                            class="MuiTouchRipple-root"
                          />
                        </button>
                      </div>
                    </div>
                    <div
                      class="MuiDataGrid-columnSeparator"
                      style="min-height: 56px; opacity: 1;"
                    >
                      <svg
                        aria-hidden="true"
                        class="MuiSvgIcon-root MuiDataGrid-iconSeparator"
                        focusable="false"
                        viewBox="0 0 24 24"
                      >
                        <path
                          d="M11 19V5h2v14z"
                        />
                      </svg>
                    </div>
                  </div>
                  <div
                    aria-colindex="2"
                    class="MuiDataGrid-columnHeader"
                    data-field="documento"
                    role="columnheader"
                    style="width: 140px; min-width: 140px; max-width: 140px;"
                    tabindex="-1"
                  >
                    <div
                      class="MuiDataGrid-columnHeaderDraggableContainer"
                      draggable="false"
                    >
                      <div
                        class="MuiDataGrid-columnHeaderTitleContainer"
                      >
                        <div
                          class="MuiDataGrid-columnHeaderTitle"
                          title=""
                        >
                          Documento
                        </div>
                      </div>
                      <div
                        class="MuiDataGrid-menuIcon"
                      >
                        <button
                          aria-controls="mui-94950"
                          aria-haspopup="true"
                          aria-label="Menu"
                          class="MuiButtonBase-root MuiIconButton-root MuiDataGrid-menuIconButton MuiIconButton-sizeSmall"
                          id="mui-9974"
                          tabindex="-1"
                          title="Menu"
                          type="button"
                        >
                          <span
                            class="MuiIconButton-label"
                          >
                            <svg
                              aria-hidden="true"
                              class="MuiSvgIcon-root MuiSvgIcon-fontSizeSmall"
                              focusable="false"
                              viewBox="0 0 24 24"
                            >
                              <path
                                d="M12 8c1.1 0 2-.9 2-2s-.9-2-2-2-2 .9-2 2 .9 2 2 2zm0 2c-1.1 0-2 .9-2 2s.9 2 2 2 2-.9 2-2-.9-2-2-2zm0 6c-1.1 0-2 .9-2 2s.9 2 2 2 2-.9 2-2-.9-2-2-2z"
                              />
                            </svg>
                          </span>
                          <span
                            class="MuiTouchRipple-root"
                          />
                        </button>
                      </div>
                    </div>
                    <div
                      class="MuiDataGrid-columnSeparator"
                      style="min-height: 56px; opacity: 1;"
                    >
                      <svg
                        aria-hidden="true"
                        class="MuiSvgIcon-root MuiDataGrid-iconSeparator"
                        focusable="false"
                        viewBox="0 0 24 24"
                      >
                        <path
                          d="M11 19V5h2v14z"
                        />
                      </svg>
                    </div>
                  </div>
                  <div
                    aria-colindex="3"
                    class="MuiDataGrid-columnHeader"
                    data-field="telefono"
                    role="columnheader"
                    style="width: 130px; min-width: 130px; max-width: 130px;"
                    tabindex="-1"
                  >
                    <div
                      class="MuiDataGrid-columnHeaderDraggableContainer"
                      draggable="false"
                    >
                      <div
                        class="MuiDataGrid-columnHeaderTitleContainer"
                      >
                        <div
                          class="MuiDataGrid-columnHeaderTitle"
                          title=""
                        >
                          Teléfono
                        </div>
                      </div>
                      <div
                        class="MuiDataGrid-menuIcon"
                      >
                        <button
                          aria-controls="mui-22604"
                          aria-haspopup="true"
                          aria-label="Menu"
                          class="MuiButtonBase-root MuiIconButton-root MuiDataGrid-menuIconButton MuiIconButton-sizeSmall"
                          id="mui-71105"
                          tabindex="-1"
                          title="Menu"
                          type="button"
                        >
                          <span
                            class="MuiIconButton-label"
                          >
                            <svg
                              aria-hidden="true"
                              class="MuiSvgIcon-root MuiSvgIcon-fontSizeSmall"
                              focusable="false"
                              viewBox="0 0 24 24"
                            >
                              <path
                                d="M12 8c1.1 0 2-.9 2-2s-.9-2-2-2-2 .9-2 2 .9 2 2 2zm0 2c-1.1 0-2 .9-2 2s.9 2 2 2 2-.9 2-2-.9-2-2-2zm0 6c-1.1 0-2 .9-2 2s.9 2 2 2 2-.9 2-2-.9-2-2-2z"
                              />
                            </svg>
                          </span>
                          <span
                            class="MuiTouchRipple-root"
                          />
                        </button>
                      </div>
                    </div>
                    <div
                      class="MuiDataGrid-columnSeparator"
                      style="min-height: 56px; opacity: 1;"
                    >
                      <svg
                        aria-hidden="true"
                        class="MuiSvgIcon-root MuiDataGrid-iconSeparator"
                        focusable="false"
                        viewBox="0 0 24 24"
                      >
                        <path
                          d="M11 19V5h2v14z"
                        />
                      </svg>
                    </div>
                  </div>
                  <div
                    class="MuiDataGrid-cell"
                    style="min-width: 620px; max-width: 620px; line-height: 55px; min-height: 56px; max-height: 56px;"
                  />
                </div>
              </div>
              <div
                style="overflow: visible; height: 0px; width: 0px;"
              >
                <div
                  class="MuiDataGrid-windowContainer"
                  style="width: 0px; height: 0px;"
                >
                  <div
                    class="MuiDataGrid-window"
                    style="top: 56px; overflow-y: auto;"
                  >
                    <div
                      class="MuiDataGrid-dataContainer"
                      style="min-width: 1010px; min-height: 1px;"
                    >
                      <div
                        class="MuiDataGrid-viewport"
                        style="min-width: 0; max-width: 0; max-height: 0;"
                      >
                        <div
                          class="MuiDataGrid-renderingZone"
                          style="max-height: 1px; width: 1010px; transform: translate3d(-0px, -0px, 0);"
                        />
                      </div>
                    </div>
                  </div>
                </div>
              </div>
              <div
                class="Mui-resizeTriggers"
              >
                <div
                  class="expand-trigger"
                >
                  <div
                    style="width: 1px; height: 1px;"
                  />
                </div>
                <div
                  class="contract-trigger"
                />
              </div>
            </div>
            <div>
              <div
                class="MuiDataGrid-footerContainer"
              >
                <div />
                <div
                  class="MuiTablePagination-root"
                >
                  <div
                    class="MuiToolbar-root MuiToolbar-regular MuiTablePagination-toolbar MuiToolbar-gutters"
                  >
                    <div
                      class="MuiTablePagination-spacer"
                    />
                    <p
                      class="MuiTypography-root MuiTablePagination-caption makeStyles-caption-4 MuiTypography-body2 MuiTypography-colorInherit"
                    >
                      0-0 of 0
                    </p>
                    <div
                      class="MuiTablePagination-actions"
                    >
                      <button
                        aria-label="Previous page"
                        class="MuiButtonBase-root MuiIconButton-root MuiIconButton-colorInherit Mui-disabled Mui-disabled"
                        disabled=""
                        tabindex="-1"
                        title="Previous page"
                        type="button"
                      >
                        <span
                          class="MuiIconButton-label"
                        >
                          <svg
                            aria-hidden="true"
                            class="MuiSvgIcon-root"
                            focusable="false"
                            viewBox="0 0 24 24"
                          >
                            <path
                              d="M15.41 16.09l-4.58-4.59 4.58-4.59L14 5.5l-6 6 6 6z"
                            />
                          </svg>
                        </span>
                      </button>
                      <button
                        aria-label="Next page"
                        class="MuiButtonBase-root MuiIconButton-root MuiIconButton-colorInherit Mui-disabled Mui-disabled"
                        disabled=""
                        tabindex="-1"
                        title="Next page"
                        type="button"
                      >
                        <span
                          class="MuiIconButton-label"
                        >
                          <svg
                            aria-hidden="true"
                            class="MuiSvgIcon-root"
                            focusable="false"
                            viewBox="0 0 24 24"
                          >
                            <path
                              d="M8.59 16.34l4.58-4.59-4.58-4.59L10 5.75l6 6-6 6z"
                            />
                          </svg>
                        </span>
                      </button>
                    </div>
                  </div>
                </div>
              </div>
            </div>
          </div>
        </div>
      </div>
    </div>
  </div>
  <form />
</div>
`;
